src/oracles/ChainlinkPriceOracle.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.8.0 <0.9.0;
+
+import {AggregatorV3Interface} from "../interfaces/AggregatorV3Interface.sol";
+import {IPriceOracle} from "../interfaces/IPriceOracle.sol";
+import {IWatchtowerCustomErrors} from "../interfaces/IWatchtowerCustomErrors.sol";
+/**
+ * @title Chainlink Price Oracle
+ * @author GUNBOATs
+ * @dev This contract creates an oracle that is compatible with the IPriceOracle
+ * interface and can be used by a CoW AMM to determine the current price of the
+ * traded tokens from Chainlink Data Feeds with decimals lower than 18.
+ */
+
+contract ChainlinkPriceOracle is IPriceOracle, IWatchtowerCustomErrors {
+    /**
+     * @param token0Feed Address of token0 oracle
+     * @param token1Feed Address of token1 oracle
+     * @param timeThreshold Amount of seconds before it consider as stale oracle
+     * @param backoff Duration to indicated how long the watchtower will wait for oracle to refresh
+     */
+    struct Data {
```

Comment:
Is there a better name than `Data` here.

Reply:
This follows the naming of other implementation in this repo,  I have no idea for name though lol.
(There is no change in code if this struct is renamed)